testlapack/dlahr2.go

```diff
@@ -0,0 +1,83 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/gonum/floats"
+)
+
+type Dlahr2er interface {
+	Dlahr2(n, k, nb int, a []float64, lda int, tau, t []float64, ldt int, y []float64, ldy int)
+}
+
+type Dlahr2test struct {
+	N, K, NB int
+	A        []float64
+
+	AWant   []float64
+	TWant   []float64
+	YWant   []float64
+	TauWant []float64
+}
+
+func Dlahr2Test(t *testing.T, impl Dlahr2er) {
```

Comment:
This is exactly what I was thinking yesterday when I started working on Dgehrd. I'll take a look at it  and see what I can do.